### ISSUE TITLE
Cookbook to converge completely on Windows Server 2012 R2

### ISSUE THREAD

**Issue by rhass-r7** [criteo-cookbooks/wsus-server#1]
The following error occurs when attempting to run the cookbook for the first time on a fresh Windows 2012R2 instance:

```
 win2012r2 [2014-10-08T16:13:27-07:00] INFO: Processing windows_feature[IIS-WebServerRole] action install (iis::default line 22)
 win2012r2 [2014-10-08T16:14:16-07:00] INFO: windows_feature[IIS-WebServerRole] installed feature
 win2012r2 [2014-10-08T16:14:16-07:00] INFO: Processing windows_feature[IIS-WebServer] action install (iis::default line 22)
 win2012r2 [2014-10-08T16:14:18-07:00] INFO: Processing windows_feature[IIS-ApplicationDevelopment] action install (iis::default line 22)
 win2012r2 [2014-10-08T16:14:20-07:00] INFO: Processing windows_feature[IIS-ISAPIFilter] action install (iis::default line 22)
 win2012r2 [2014-10-08T16:14:28-07:00] INFO: windows_feature[IIS-ISAPIFilter] installed feature
 win2012r2 [2014-10-08T16:14:28-07:00] INFO: Processing windows_feature[IIS-ISAPIExtensions] action install (iis::default line 22)
 win2012r2 [2014-10-08T16:14:37-07:00] INFO: windows_feature[IIS-ISAPIExtensions] installed feature
 win2012r2 [2014-10-08T16:14:37-07:00] INFO: Processing windows_feature[IIS-NetFxExtensibility] action install (iis::default line 22)
 win2012r2 
 win2012r2 ================================================================================
 win2012r2 Error executing action `install` on resource 'windows_feature[IIS-NetFxExtensibility]'
 win2012r2 ================================================================================
 win2012r2 
 win2012r2 Mixlib::ShellOut::ShellCommandFailed
 win2012r2 ------------------------------------
 win2012r2 Expected process to exit with [0, 42, 127, 3010], but received '50'
 win2012r2 ---- Begin output of C:\Windows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   ----
 win2012r2 STDOUT: Deployment Image Servicing and Management tool
 win2012r2 Version: 6.3.9600.16384
 win2012r2 
 win2012r2 Image Version: 6.3.9600.16384
 win2012r2 
 win2012r2 Enabling feature(s)
 win2012r2 
 win2012r2 Error: 50
 win2012r2 
 win2012r2 The operation is complete but IIS-NetFxExtensibility feature was not enabled.
 win2012r2 A required parent feature may not be enabled. You can use the /enable-feature /all option to automatically enable each parent feature from the following list. If the parent feature(s) are already enabled, refer to the log file for further diagnostics.
 win2012r2 NetFx3, NetFx3ServerFeatures, NetFx4Extended-ASPNET45
 win2012r2 
 win2012r2 The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
 win2012r2 STDERR: 
 win2012r2 ---- End output of C:\Windows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   ----
 win2012r2 Ran C:\Windows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   returned 50
 win2012r2 
 win2012r2 Cookbook Trace:
 win2012r2 ---------------
 win2012r2 c:/chef/cache/cookbooks/windows/providers/feature_dism.rb:28:in `install_feature'
 win2012r2 c:/chef/cache/cookbooks/windows/libraries/feature_base.rb:8:in `action_install'
 win2012r2 
 win2012r2 Resource Declaration:
 win2012r2 ---------------------
 win2012r2 # In c:/chef/cache/cookbooks/iis/recipes/default.rb
 win2012r2 
 win2012r2  22:   windows_feature feature do
 win2012r2  23:     action :install
 win2012r2  24:   end
 win2012r2  25: end
 win2012r2 
 win2012r2 Compiled Resource:
 win2012r2 ------------------
 win2012r2 # Declared in c:/chef/cache/cookbooks/iis/recipes/default.rb:22:in `block in from_file'
 win2012r2 
 win2012r2 windows_feature("IIS-NetFxExtensibility") do
 win2012r2   provider Chef::Provider::WindowsFeatureDism
 win2012r2   action [:install]
 win2012r2   retries 0
 win2012r2   retry_delay 2
 win2012r2   guard_interpreter :default
 win2012r2   cookbook_name "iis"
 win2012r2   recipe_name "default"
 win2012r2   feature_name "IIS-NetFxExtensibility"
 win2012r2 end
 win2012r2 
 win2012r2 [2014-10-08T16:14:41-07:00] INFO: Running queued delayed notifications before re-raising exception
 win2012r2 [2014-10-08T16:14:41-07:00] ERROR: Running exception handlers
 win2012r2 [2014-10-08T16:14:41-07:00] ERROR: Exception handlers complete
 win2012r2 [2014-10-08T16:14:41-07:00] FATAL: Stacktrace dumped to c:/chef/cache/chef-stacktrace.out
 win2012r2 [2014-10-08T16:14:41-07:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: windows_feature[IIS-NetFxExtensibility] (iis::default line 22) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0, 42, 127, 3010], but received '50'
 win2012r2 ---- Begin output of C:\Windows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   ----
 win2012r2 STDOUT: Deployment Image Servicing and Management tool
 win2012r2 Version: 6.3.9600.16384
 win2012r2 
 win2012r2 Image Version: 6.3.9600.16384
 win2012r2 
 win2012r2 Enabling feature(s)
 win2012r2 
 win2012r2 Error: 50
 win2012r2 
 win2012r2 The operation is complete but IIS-NetFxExtensibility feature was not enabled.
 win2012r2 A required parent feature may not be enabled. You can use the /enable-feature /all option to automatically enable each parent feature from the following list. If the parent feature(s) are already enabled, refer to the log file for further diagnostics.
 win2012r2 NetFx3, NetFx3ServerFeatures, NetFx4Extended-ASPNET45
 win2012r2 
 win2012r2 The DISM log file can be found at C:\Windows\Logs\DISM\dism.log
 win2012r2 STDERR: 
 win2012r2 ---- End output of C:\Win
 win2012r2 dows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   ----
 win2012r2 Ran C:\Windows\sysnative\dism.exe /online /enable-feature /featurename:IIS-NetFxExtensibility /norestart   returned 50
```

**Comment by Annih** [comment]
Hi  @rhass-r7  and @svmastersamurai 

Sorry for the delay, I was very busy recently, but I did work on this issue.
I currently have a local branch with an improved support of Windows server 2012R2 ...
Actually the support was inexistent yet, because it tried to install package instead of using the WSUS Role ...
Stay tuned, I didn't forget you! I'll share my work in the next days :)

**Comment by rhass-r7** [comment]
That is excellent news! Thanks for the update! =)

**Comment by dschrander** [comment]
Running into the same issue here. Kinda stuck. Any updates you can share already? Maybe a push in the right direction regarding solving it? Thanks!

**Comment by Annih** [comment]
Hi,

Again, sorry for the delay, we just released the version 1.0.0 of the cookbook wsus-server.
It properly supports Windows server 2012.
Let me know if you like it :)
